### PR TITLE
Sanitize setters

### PR DESCRIPTION
**Pull request by gzm0**

### CODE REVIEW

**Comment by gzm0** [comment]
Review by @sjrd. Note: I have been a bit more aggressive on disallowing things.

File: compiler/src/main/scala/org/scalajs/core/compiler/PrepJSExports.scala [hunk 1]
```diff
@@ -189,13 +189,9 @@ trait PrepJSExports { this: PrepJSInterop =>
...
         else sym.unexpandedName.decoded.stripSuffix("_=")
 
-      // Enforce that methods ending with _= are exported as setters
-      if (sym.isMethod && !sym.isConstructor &&
-        sym.name.decoded.endsWith("_=") && !jsInterop.isJSSetter(sym)) {
-        reporter.error(annot.pos, "A method ending in _= will be exported " +
-            s"as setter. But ${sym.name.decoded} does not have the right " +
-            "signature to do so (single argument, unit return type).")
-      }
+      // Enforce that setter properties
```
Comment (sjrd) [inline compiler/src/main/scala/org/scalajs/core/compiler/PrepJSExports.scala]: This sentence seems unfinished.

File: compiler/src/main/scala/org/scalajs/core/compiler/PrepJSInterop.scala [hunk 2]
```diff
@@ -528,6 +525,38 @@ abstract class PrepJSInterop extends plugins.PluginComponent
   def isJSAny(sym: Symbol): Boolean =
     sym.tpe.typeSymbol isSubClass JSAnyClass
 
+  /** Checks that a setter has the right signature
+   *
+   *  Reports error messages otherwise
```
Comment (sjrd) [inline compiler/src/main/scala/org/scalajs/core/compiler/PrepJSInterop.scala]: These two sentences lack a trailing `.`

**Comment by sjrd** [comment]
That's all.

File: compiler/src/main/scala/org/scalajs/core/compiler/PrepJSInterop.scala [hunk 3]
```diff
@@ -528,6 +525,38 @@ abstract class PrepJSInterop extends plugins.PluginComponent
...
+    sym.tpe.paramss match {
+      case List(List(arg)) =>
+        // Arg list is OK. Do additional checks.
+        if (isScalaRepeatedParamType(arg.tpe))
+          reporter.error(pos, s"$typeStr setters may not have repeated params")
+
+        if (arg.hasFlag(reflect.internal.Flags.DEFAULTPARAM)) {
+          val msg = s"$typeStr setters may not have default params"
+          if (exported)
+            reporter.warning(pos, msg)
```
Comment (gzm0) [inline compiler/src/main/scala/org/scalajs/core/compiler/PrepJSInterop.scala]: This should have a trailing "This will be enforced in 1.0."

**Comment by gzm0** [comment]
Updated.

**Comment by sjrd** [comment]
LGTM

**Pull request merged**